### PR TITLE
Model config

### PR DESCRIPTION
**Pull request by evilnick**
Fixes #987

### CODE REVIEW

File: src/en/models-config.md [hunk 1]
```diff
@@ -120,37 +167,11 @@ enable-os-upgrade: false
...
-```
-
-It is not desirable to set this feature on hardware which is not NUMA
-compatible.
-
-
 ## Juju lifecycle and harvesting
 
-Juju keeps state on the running environment from which it builds a model.
+Juju keeps state on the running model from which it builds a model.
```
Comment (pmatulis) [inline src/en/models-config.md]: model model

File: src/en/models-config.md [hunk 2]
```diff
@@ -1,72 +1,135 @@
...
+juju unset-model-config apt-mirror
 ```
-to restore the default behaviour in a running environment.
+to restore the default behaviour in a running model.
 
 
 ## Versions and Streams
 
-The ```agent-stream``` option selects the versions of Juju which an environment
+The ```agent-stream``` option selects the versions of Juju which an model
```
Comment (pmatulis) [inline src/en/models-config.md]: a model

File: src/en/models-config.md [hunk 3]
```diff
@@ -1,72 +1,135 @@
...
 
 ```bash
-juju set-env noproxy=jujucharms.com
+juju set-model-config noproxy=jujucharms.com
+```
+It is possible to specify a list of space-delimited key-value pairs to set more
+than one configuration at a time:
+  
+```bash
+juju set-model-config test-mode=true enable-os-upgrade=false
```
Comment (pmatulis) [inline src/en/models-config.md]: I see this:

$ juju get-model-config | grep refresh
$ juju set-model-config enable-os-refresh-update=false
WARNING key "enable-os-refresh-update" is not defined in the current model configuration: possible misspelling
$ juju get-model-config | grep refresh
enable-os-refresh-update: false

Is this what we want?

File: src/en/models-config.md [hunk 4]
```diff
@@ -1,72 +1,135 @@
...
 or running multiple instances of Juju, where it would be useful to control
-further aspects of the Juju environment. Many options can be changed by adding
-extra key:value pairs to the environments.yaml file which Juju uses to initiate
-environments. Options which are specific to certain cloud providers are
-detailed in the individual configuration pages, but there are additional
-settings which can be applied to any environment. Some of these may be useful
-for particular use cases as outlined in the sections below, or you can
-[skip to the table](#alphabetical-list-of-general-configuration-values) of all
-the values.
+further aspects of the models you create.
```
Comment (pmatulis) [inline src/en/models-config.md]: may want to state explicitly that any machines created in the configured model will inherit these settings

**Comment by pmatulis** [comment]
Review complete

File: src/en/models-config.md (hunk 3 above)
Comment (evilnick) [inline src/en/models-config.md]: Not sure what you mean. If you mean, do we want a scary error message every time we add some key that isn't there by default - see https://bugs.launchpad.net/juju-core/+bug/1413653

File: src/en/models-config.md (hunk 4 above)
Comment (evilnick) [inline src/en/models-config.md]: the machines are created 'with these' settings, but they don't really inherit them. Anyhow, I shall reword

File: src/en/models-config.md (hunk 3 above)
Comment (pmatulis) [inline src/en/models-config.md]: yeah, the bug covers it. warning looks like i did something wrong and the action did not take place

**Pull request merged**